lib/services/matchers.go

```diff
@@ -37,9 +44,13 @@ type AWSMatcher struct {
 	Regions []string
 	// Tags are AWS tags to match.
 	Tags types.Labels
-	// SSMDocument is the SSM document used to execute the
-	// installation script
-	SSMDocument string
+	// JoinMethod is the join method to use when discovering an EC2 node
+	JoinMethod types.JoinMethod
+	// JoinToken is the token to use when discovering an EC2 node
+	JoinToken string
+	// SSM provides options to use when sending a document command to
+	// an EC2 node
+	SSM *AWSSSM `yaml:"ssm,omitempty"`
```

Comment:
Does this field need a json tag? Others don't have it (I don't think this struct is marshaled/unmarshaled).